docs/writing/posts/retrieval-is-more-than-pure-chance.md

```diff
@@ -0,0 +1,675 @@
+| 44b1f682be6d29b399b2947a426137e8 | What should you do if you're young and ambitious but don't know what to work on? What you should not do is drift along passively, assuming the problem will solve itself. You need to take action. But there is no systematic procedure you can follow. When you read biographies of people who've done great work, it's remarkable how much luck is involved. They discover what to work on as a result of a chance meeting, or by reading a book they happen to pick up. So you need to make yourself a big target for luck, and the way to do that is to be curious. Try lots of things, meet lots of people, read lots of books, ask lots of questions. |
+| 5ff0ae69052dccc9af9cb6a211dfe127 | The first step is to decide what to work on. The work you choose needs to have three qualities: it has to be something you have a natural aptitude for, that you have a deep interest in, and that offers scope to do great work. |
+| bf23afd092d5c6d8179c72d2ab7f1d46 | The factors in doing great work are factors in the literal, mathematical sense, and they are: ability, interest, effort, and luck. Luck by definition you can't do anything about, so we can ignore that. And we can assume effort, if you do in fact want to do great work. So the problem boils down to ability and interest. Can you find a kind of work where your ability and interest will combine to yield an explosion of new ideas? |
+| 87aa6353fe45ce7dae2b7d6e42d66733 | Fortunately there's a kind of economy of scale here. Though it might seem like you'd be taking on a heavy burden by trying to be the best, in practice you often end up net ahead. It's exciting, and also strangely liberating. It simplifies things. In some ways it's easier to try to be the best than to try merely to be good. |
+
+
+And with that, we've figured out how to implement retrieval
+
+## Evaluations
+
+There are two useful evaluations which we can use when evaluating the quality of retrievals - NDCG and MRR. Intuitively, since we used a single chunk to generate a question, we want to measure how often it appears as the top result and its overall ranking when we fetch `k` different chunks.
+
+Eventually, as we utilise more complex methods to generate our questions - either by combining multiple chunks or writing more complex queries, our approach will change. But for now, these two metrics are great for quickly determining how good our retrieval is performing.
+
+??? info "Evaluations"
+    
+    For a more complex and in-depth look into different metrics, consider looking at [this article written by Jason on evaluating Machine Learning Systems](https://jxnl.github.io/blog/writing/2024/02/05/when-to-lgtm-at-k/#mean-reciprocal-rank-mrr-k)
+
+Thankfully, we have a native implementation of NDCG in sklearn and MRR is relatively simple to implement. Let's see how we can calculate these metrics
+
+### Metrics
+
+#### NDCG
```

Comment:
we already talked about metrics in a previous post just link to it and give a short overview